test/unit/services/validation.service.test.ts

```diff
@@ -129,4 +172,80 @@ describe('ValidationService', () => {
       }
     });
   });
+
+  describe('validateDocument() with external URLs', () => {
+    let originalConfig: any;
+    const CONFIG_DIR = path.join(os.homedir(), '.asyncapi');
+    const CONFIG_FILE = path.join(CONFIG_DIR, 'config.json');
+
+    beforeEach(async () => {
+      // Backup original config
+      try {
+        const content = await fs.readFile(CONFIG_FILE, 'utf8');
+        originalConfig = JSON.parse(content);
+      } catch (err) {
+        originalConfig = null;
+      }
+    });
+
+    afterEach(async () => {
+      // Restore original config
+      if (originalConfig) {
+        await fs.writeFile(CONFIG_FILE, JSON.stringify(originalConfig, null, 2), 'utf8');
+      } else {
+        try {
+          await fs.unlink(CONFIG_FILE);
+        } catch (err) {
+          // File doesn't exist, ignore
+        }
+      }
+    });
+
+    it('should validate document with private GitHub reference', async () => {
+      // Ensure no auth config exists for private GitHub repository
+      try {
+        await fs.unlink(CONFIG_FILE);
+      } catch (err) {
+        // File doesn't exist, ignore
+      }
+
+      const specFile = new Specification(asyncAPIWithPrivateGitHubRef);
+      const options = {
+        'diagnostics-format': 'stylish' as const
+      };
+
+      const result = await validationService.validateDocument(specFile, options);
+
+      // The validation succeeds but the document is invalid due to unresolved ref
+      expect(result.success).to.equal(true);
```

Comment:
If this test is passing we're doing something wrong. `result.success` shouldn't be `true` because a key part of the document couldn't even be resolved.

Reply:
Don't you find it weird that `success === true` but the document is invalid?